libsolidity/interface/CompilerStack.cpp

```diff
@@ -433,7 +434,7 @@ void CompilerStack::link()
 
 vector<string> CompilerStack::contractNames() const
 {
-	if (m_stackState < AnalysisSuccessful)
+	if (m_stackState < AnalysisSuccessful && !m_parserErrorRecovery)
```

Comment:
I think there has to be a better way that just appending `&& !m_parserErrorRecovery` everywhere

Reply:
I have looked at this, and as far as I can tell, there is only this one place. 

I am stuck here. If you can come up code that is cleaner please make the change. Thanks.